opentelemetry-sdk/src/metrics/internal/histogram.rs

```diff
@@ -22,27 +18,26 @@ where
     type PreComputedValue = (T, usize);
 
     fn update(&self, (value, index): (T, usize)) {
-        let mut buckets = match self.buckets.lock() {
-            Ok(guard) => guard,
-            Err(_) => return,
-        };
+        let mut buckets = self.lock().unwrap_or_else(|err| err.into_inner());
```

Comment:
perhaps a good idea to start using internal logs for these errors?

Reply:
just to confirm I understand this right:
These locks are taken by user-threads when calling histogram.Record(..).
But the only code executed within the lock is these 4 lines to update min/max/sum/count, which has no chance of panic.
And due to that, we don't need to worry about poisoning here.

The other place where this lock is taken is the collect() thread, but that is only executing a mem-replace, with no possibility of panic.

And because of this, we don't ever expect a lock poison here?

Is there a possibility that the user thread after acquiring the lock gets somehow killed? I am not saying we should account for that here!

Reply:
> But the only code executed within the lock is these 4 lines to update min/max/sum/count, which has no chance of panic.
And due to that, we don't need to worry about poisoning here.

Updating sum or count could lead to overflows. Even though vverflows would cause a panic only in debug build, I think we should update these operations to always use `wrapping_add`.

> The other place where this lock is taken is the collect() thread, but that is only executing a mem-replace, with no possibility of panic.

Currently, the replace involves creating a new set of buckets which needs allocation. The worst case is we run into OOM doing that but if that happens, we have bigger problems than lock poisoning 😄

> Is there a possibility that the user thread after acquiring the lock gets somehow killed?

I'm not sure if that is possible.

Reply:
> Currently, the replace involves creating a new set of buckets which needs allocation. The worst case is we run into OOM doing that but if that happens, we have bigger problems than lock poisoning 

Once we fix the logic to use pre-allocated one, this concern should be mitigated right?

I'll resolve this comment to not block.

Reply:
Yes, it should be fixable.

Using `wrapping_add` instead of regular arithmetic operations can be done outside of this PR.

Reply:
> Is there a possibility that the user thread after acquiring the lock gets somehow killed? I am not saying we should account for that here!

AFAIK this is not possible.

Reply:
> Yes, it should be fixable.
> 
> Using `wrapping_add` instead of regular arithmetic operations can be done outside of this PR.

Next PR is coming soon, which will make that 'clone_and_reset' will be used only by no_attributes_tracker.

Regarding wrapping_add, its nice idea:)